rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -859,6 +855,55 @@ public void onSuccess(String result) {
+     *
+     * @param success call this when it's done
+     * @param failure call this if it fails
+     */
+    public void fetchNodeSourceConfiguration(String nodeSourceName, Runnable success, Runnable failure) {
+
+        rm.getNodeSourceConfiguration(LoginModel.getInstance().getSessionId(),
+                                      nodeSourceName,
+                                      new AsyncCallback<String>() {
+
+                                          public void onSuccess(String result) {
+                                              model.setEditedNodeSourceConfiguration(parseNodeSourceConfiguration(result));
+                                              success.run();
+                                          }
+
+                                          public void onFailure(Throwable caught) {
+                                              String msg = JSONUtils.getJsonErrorMessage(caught);
+                                              SC.warn("Failed to fetch configuration of node source " + nodeSourceName +
+                                                      ":<br>" + msg);
+                                              failure.run();
+                                          }
+                                      });
+    }
+
+    private NodeSourceConfiguration parseNodeSourceConfiguration(String json) {
+
+        JSONObject jsonObject = this.parseJSON(json).isObject();
+
+        String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
+
+        boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
+
+        JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+        String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName").isString().stringValue();
+        PluginDescriptor infrastructurePluginDescriptor = getPluginDescriptor(infrastructurePluginDescriptorJson,
+                                                                              infrastructurePluginName);
+
+        JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
+        String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
+        PluginDescriptor policyPluginDescriptor = getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+
+        return new NodeSourceConfiguration(nodeSourceName,
+                                           nodesRecoverable,
+                                           infrastructurePluginDescriptor,
+                                           policyPluginDescriptor);
+    }
+
     private HashMap<String, PluginDescriptor> parsePluginDescriptors(String json) {
         JSONArray arr = this.parseJSON(json).isArray();
         HashMap<String, PluginDescriptor> plugins = new HashMap<String, PluginDescriptor>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)